cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go

```diff
@@ -335,6 +335,7 @@ func TestValidateMasterConfiguration(t *testing.T) {
 				Networking: kubeadm.Networking{
 					ServiceSubnet: "10.96.0.1/12",
 					DNSDomain:     "cluster.local",
+					PodSubnet:     "10.0.1.15/16",
```

Comment:
This value added, but seems to be neither used nor affects the test case. Why it is added ?

Reply:
I was thinking of adding this to cover the below condition:
```go
	if len(c.PodSubnet) != 0 {
		allErrs = append(allErrs, ValidateIPNetFromString(c.PodSubnet, constants.MinimumAddressesInServiceSubnet, field.NewPath("pod-subnet"))...)
	}
```
https://github.com/xiangpengzhao/kubernetes/blob/b09bfb481ac1c60dba50ab6c51cddd4c616b6b82/cmd/kubeadm/app/apis/kubeadm/validation/validation.go#L252-L254

Reply:
Well, for testing that it will be better to have two separate items: one with valid PodSubnet, one with invalid PodSubnet to trigger error. Same applies to ServiceSubnet: we have positive cases, good to add negative case to cover fully `ValidateNetworking()`

Reply:
Because we already have `TestValidateIPFromString` to test (cover) the invalid cases of `PodSubnet` so I didn't add negative case here. Do I still add it here?

Reply:
My point was more about testing function `ValidateNetworking()` itself. `TestValidateIPFromString` and `TestValidateIPNetFromString` are validating that basic network functions works. Here we can make sure that whole path from top to bottom works by test case that will trigger different code paths in `ValidateNetworking`.